Fix to support sparse column multi-index + test case.

yohplala opened the pull request:
Fixing support of sparse column mutli-index, as mentionned in ticket #728

martindurant commented on fastparquet/writer.py:
```diff
@@ -428,15 +434,12 @@ def write_column(f, data, selement, compression=None, datapage_version=None,
         type to use, which must be one of the keys in ``compression.compress``,
         and may optionally have key ``"args`` which should be a dictionary of
         options to pass to the underlying compression engine.
-<<<<<<< HEAD
```
Ooops :)

martindurant commented:
Thanks, @yohplala . I'll look sometime next week. 
The critical thing with this is to cover all the cases we want to handle in the test. Or else, decide on which cases we decide *not* to handle and raise exceptions for those as needed.

yohplala commented:
> I'll look sometime next week.

Thanks @martindurant 

>  The critical thing with this is to cover all the cases we want to handle in the test. Or else, decide on which cases we decide _not_ to handle and raise exceptions for those as needed.

Please, here are some thoughts on this topic.

I have added the example above, which for one column and one level, has an empty string as part of the column name.
I checked and `norm_col_name()` was added for handling column multi-index. I understand it has been added for a reason, but I am not sure to understand which one.
I can only think about the row index case, meaning when the row index is reset by fastparquet, it makes no sense to keep the empty string.
In a previous version of the fix, I removed `norm_col_name` and the modified index name would find itself in the output dataframe, so I can understand we do not want keeping the empty strings for it.
As a result, in proposed fix, the processing is kept for row index.

For the column names, they are user defined.
The rationale makes sense I can think is that we should not modify user data.

The user could for instance:
```python
# With 2nd column having sparse name, wherever the empty string(s) being.
writer.write(tempdir, df, file_scheme='hive')
col2 = df.columns[1]    # so, for instance "('ts', '')"
```
Then, reload data later on:
```python
df_out = pf.to_pandas(columns=[col2])
df_out      # with dataframe given above
Out[9]:
component                  ts
point                        
0         2021-01-01 08:00:00
1         2021-01-05 10:00:00
```

Now, as you point out, sparse multi-column index could also puposedly not being supported by fastparquet. But I have not identified reason not to.

martindurant commented:
Thanks for this one, I think it's good.
